internal/lsp/cache/cache.go

```diff
@@ -84,7 +84,12 @@ func (c *Cache) GetAllFiles() map[string]string {
 	c.fileContentsMu.Lock()
 	defer c.fileContentsMu.Unlock()
 
-	return c.fileContents
+	cp := make(map[string]string, len(c.fileContents))
```

Comment:
I believe we could use [`maps.Clone`](https://pkg.go.dev/golang.org/x/exp/maps#Clone) for this

Reply:
Ohh go is cool now.

Reply:
I have added this in 51758077f96655a00f3ec60bc68e0d5f98fceac3